server/test/upload-clients.js

```diff
@@ -0,0 +1,40 @@
+import fs from "fs";
+import MindbodyRequest from "../requests.js";
+import MindbodyAccess from "../api-manager.js";
+
+const URL = "https://api.mindbodyonline.com/public/v6/client";
+const APIKEY = "76af57a017f64fcd9fc16cc5032404a0";
+const SITEID = "-99";
+
+const fileName = process.argv.slice(2)[0];
+
+function addClient(parameters) {
+    var request = new MindbodyRequest(
+        URL + "/addclient",
+        APIKEY,
+        SITEID,
+        "POST",
+        parameters,
+    );
+    request.addAuth(MindbodyAccess.authToken);
+    return request.makeRequest();
+}
+
+fs.readFile('./server/test/' + fileName, 'utf8', (err, jsonString) => {
+    if (err) {
+        console.log("File read failed:", err)
+        return
+    }
+    try {
+        const clientsList = JSON.parse(jsonString)
+        MindbodyAccess.getAuth()
+            .then((value) => {
+                MindbodyAccess.authToken = value.AccessToken;
+            });
+        for (let i = 0; i < clientsList['clients'].length; i++) {
+            addClient(clientsList['clients'][i]);
+        }
```

Comment:
I'm NOT an expert on JS and Promises, so don't take my word for this. I think this code can make the addClient requests before its received the auth token. If so, I'd expect the addClient requests to fail because of some "bad Auth Token" error. The way I'd do it is:
        MindbodyAccess.getAuth()
            .then((value) => {
                MindbodyAccess.authToken = value.AccessToken;
                for (let i = 0; i < clientsList['clients'].length; i++) {
                    addClient(clientsList['clients'][i]);
                }
                //... and in fact if the caller wants to KNOW when the clients have been asynchronously added,
                //... then you'd need to construct a new Promise.all() using all of the promises returned by addClient
            }).catch(error => {
                //...handle errors from where these promises (getAuth mostly) were rejected
            });

Reply:
^^^ well, the indentation looks terrible in this editor. Sorry. There's some way to suggest code changes directly in the git web UI, but I can't find it.